### ISSUE TITLE
Integration Tests for MacOS

### ISSUE THREAD

**Issue by tegefaulkes** [MatrixAI/Polykey-CLI#12]
### Specification

This issue is to track the progress of PKG integration tests for the Mac build. 

A selection of bin tests need to be updated to support using the mac executable as the target for testing. 

The main tests that need to be updated in order are;
1. `bin/agent/start.test.ts`
2. `bin/bootstrap.test.ts`
3. rest of `bin/agent`
4. `bin/polykey.test.ts`

### Additional context

- Related MatrixAI/Polykey#401 

### Tasks

1. spec this out some more

**Comment by CMCDragonkai** [comment]
The current release builds in https://github.com/MatrixAI/Polykey-CLI/releases/tag/v0.18.13 says it has a `-universal` build for MacOS.

But it's a LIE!.

It's in fact not universal, it's only an arm64 build.

This means the docs is also lieing..

![Image](https://github.com/user-attachments/assets/c41fd3f9-5da0-4203-b544-af0fb67bcf12)

So basically arm64 only works on the new arm64 computers m1,m2,m3... etc.

It does not work on older intel Macs.

Do note that if it was intel, rosetta would be able to run older binaries.

The proper way to fix this is to:

1. Actually get a universal build with a fat node runtime binary.
2. Release 2 macos builds `-arm64` and `-x64`.

I guess some people are still on old macs.

**Comment by CMCDragonkai** [comment]
@xrissoula can you fix up the docs.

@aryanjassal should fix up the build pipeline here.

**Comment by nishinishic** [comment]
The install does not proceed - returned with bash: 
```
Bad CPU type in executable
```
(Intel Core i9)

**Comment by CMCDragonkai** [comment]
There is an interesting internet architecture to test on.

1. It's in China
2. It's going over a proxy that routes everything.

So I'm not sure this means there 2-layers of NAT occurring, or even 3 layers of NAT. Like I don't know if this proxy NATting as well, or if it is like a DMZ back to the Chinese router. And I'm assuming all IPs are IPv6 as well right?

Questions would be:

1. How does the China Telecom - do residential home routing - is it just local NAT plus CGNAT or only local NAT which is more common in residential homes?
2. Is it only ipv6?
3. How does a proxy impact the usage of Polykey's network.
4. Does the proxy fully take over the entire network including TCP and UDP stack?

**Comment by CMCDragonkai** [comment]
It would be interesting - if Polykey can work internally within the China network, or if with the combination of a proxy, can it also connect to our seed nodes, and therefore the rest of the network - possibly without a relay in between?

Anyway, a relay is something we need to organise too.

**Comment by tegefaulkes** [comment]
Looking at the build logs it's trying to bundle these assets in the packaged executable.

```
'node_modules/@matrixai/db/prebuilds/darwin-x64+arm64/node.napi.node',
    'node_modules/@matrixai/db/prebuilds/linux-x64/node.napi.node',
    'node_modules/@matrixai/db/prebuilds/win32-x64/node.napi.node',
    'node_modules/@matrixai/mdns-linux-x64/node.napi.node',
    'node_modules/@matrixai/mdns-linux-x64/package.json',
    'node_modules/@matrixai/quic-darwin-universal/node.napi.node',
    'node_modules/@matrixai/quic-darwin-universal/package.json',
    'node_modules/@matrixai/quic-linux-x64/node.napi.node',
    'node_modules/@matrixai/quic-linux-x64/package.json',
    'node_modules/@matrixai/quic-win32-x64/node.napi.node',
    'node_modules/@matrixai/quic-win32-x64/package.json',
    'node_modules/@matrixai/exec-darwin-universal/node.napi.node',
    'node_modules/@matrixai/exec-darwin-universal/package.json',
    'node_modules/@matrixai/exec-linux-x64/node.napi.node',
    'node_modules/@matrixai/exec-linux-x64/package.json',
    'node_modules/fd-lock/prebuilds/linux-x64/node.napi.node',
    'node_modules/fd-lock/prebuilds/darwin-x64/node.napi.node',
    'node_modules/fd-lock/prebuilds/darwin-arm64/node.napi.node',
    'node_modules/fd-lock/prebuilds/win32-x64/node.napi.node',
    'node_modules/sodium-native/prebuilds/darwin-arm64/node.napi.node',
    'node_modules/sodium-native/prebuilds/darwin-x64/node.napi.node',
    'node_modules/sodium-native/prebuilds/linux-x64/node.napi.node',
    'node_modules/sodium-native/prebuilds/win32-x64/node.napi.node',
    'node_modules/@matrixai/db/prebuilds/darwin-arm64/node.napi.node',
    'node_modules/fd-lock/prebuilds/darwin-arm64/node.napi.node',
    'node_modules/sodium-native/prebuilds/darwin-arm64/node.napi.node'
```

These are all the darwin specific paths.

```
'node_modules/@matrixai/db/prebuilds/darwin-x64+arm64/node.napi.node',
'node_modules/@matrixai/quic-darwin-universal/node.napi.node',
'node_modules/@matrixai/quic-darwin-universal/package.json',
'node_modules/@matrixai/exec-darwin-universal/node.napi.node',
'node_modules/@matrixai/exec-darwin-universal/package.json',
'node_modules/fd-lock/prebuilds/darwin-x64/node.napi.node',
'node_modules/fd-lock/prebuilds/darwin-arm64/node.napi.node',
'node_modules/sodium-native/prebuilds/darwin-arm64/node.napi.node',
'node_modules/sodium-native/prebuilds/darwin-x64/node.napi.node',
'node_modules/@matrixai/db/prebuilds/darwin-arm64/node.napi.node',
'node_modules/fd-lock/prebuilds/darwin-arm64/node.napi.node',
'node_modules/sodium-native/prebuilds/darwin-arm64/node.napi.node'
```

You can see it either provides `x64` and `arm64` native modules, or a universal binary of the two.

I'm a little confused what the problem is. It *should* be supported. is there a specific error as an example of it not working?

**Comment by CMCDragonkai** [comment]
Yes it's definitely not working because we tested the binary and read it's details, and it presents as arm64. **Having both native modules does not mean it's a universal binary. To be a TRUE universal binary, the node runtime itself HAS to be a fat binary, which it is not atm.**

We know this already because the node runtime distributions provided by PKG isn't a fat binary. I looked at the download list at yao-pkg.

![Image](https://github.com/user-attachments/assets/3a3aacc8-d048-46c9-a96d-3b9b6d097b91)
